chainstate/src/detail/tests/normal_tokens.rs

```diff
@@ -0,0 +1,1558 @@
+                            token_ticker: b"SECND".to_vec(),
+                            amount_to_issue: ISSUED_FUNDS,
+                            number_of_decimals: 1,
+                            metadata_uri: b"https://some_site.meta".to_vec(),
+                        }),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .add_output(TxOutput::new(
+                        OutputValue::Coin(token_min_issuance_fee),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process()
+            .unwrap()
+            .unwrap();
+
+        let block = tf.block(*block_index.block_id());
+        let second_issuance_outpoint_id = TestBlockInfo::from_block(&block).txns[0].0.clone();
+        let _ = token_id(&block.transactions()[0]).unwrap();
+
+        // Try to spend sum of input tokens
+
+        let token_min_issuance_fee = tf.chainstate.chain_config.token_min_issuance_fee();
+        let result = tf
+            .make_block_builder()
+            .add_transaction(
+                TransactionBuilder::new()
+                    .add_input(TxInput::new(
+                        second_issuance_outpoint_id.clone(),
+                        0,
+                        InputWitness::NoSignature(None),
+                    ))
+                    .add_input(TxInput::new(
+                        second_issuance_outpoint_id.clone(),
+                        1,
+                        InputWitness::NoSignature(None),
+                    ))
+                    .add_input(TxInput::new(
+                        second_issuance_outpoint_id,
+                        2,
+                        InputWitness::NoSignature(None),
+                    ))
+                    .add_output(TxOutput::new(
+                        OutputValue::Token(TokenData::TokenTransferV1 {
+                            token_id: first_token_id,
+                            amount: (ISSUED_FUNDS * 2).unwrap(),
+                        }),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .add_output(TxOutput::new(
+                        OutputValue::Coin((token_min_issuance_fee * 2).unwrap()),
+                        OutputPurpose::Transfer(Destination::AnyoneCanSpend),
+                    ))
+                    .build(),
+            )
+            .build_and_process();
+
+        assert!(matches!(
+            result,
+            Err(BlockError::StateUpdateFailed(
+                ConnectTransactionError::AttemptToPrintMoney(_, _)
+            ))
+        ));
+    })
+}
+
+#[rstest]
+#[trace]
+#[case(Seed::from_entropy())]
+fn test_tokens_reorgs_and_cleanup_data(#[case] seed: Seed) {
+    utils::concurrency::model(move || {
+        const ISSUED_FUNDS: Amount = Amount::from_atoms(1_000_000);
```

Comment:
Why can't this be a random number in a specified range?

Reply:
As we spoke in the chat, I added random variables where it is possible